src/server/game/Entities/Unit/Unit.h

```diff
@@ -1866,8 +1939,10 @@ class TC_GAME_API Unit : public WorldObject
         void _ExitVehicle(Position const* exitPosition = nullptr);
         void _EnterVehicle(Vehicle* vehicle, int8 seatId, AuraApplication const* aurApp = nullptr);
 
-        void BuildMovementPacket(ByteBuffer* data) const;
-        static void BuildMovementPacket(Position const& pos, Position const& transportPos, MovementInfo const& movementInfo, ByteBuffer* data);
+        MovementInfo GetMovementInfo() const;
```

Comment:
Non-virtual function with the same name in Object and Unit - will behave differently depending on what kind of pointer they are called with (Object* vs Unit*)

Unit* someUnit = ...;
someUnit->GetMovementInfo() // calls Unit::
Object* o = someUnit;
o->GetMovementInfo() // calls Object::